pkg/cloudprovider/aws/packing/packing.go

```diff
@@ -27,10 +31,14 @@ import (
 )
 
 type podPacker struct {
-	// TODO use this ec2 API to get the instance types
 	ec2 ec2iface.EC2API
 }
 
+type packingResult struct {
```

Comment:
I'm having a little trouble understanding this struct - it seems very close to a `Packing` (but doesn't include a `Packing`). It's also not always consistently returned by the "helper" functions. E.g., `packPodsForCapacity` returns it but `packWithLargestPod` does not (and, naively at least, it's not 100% clear why). I'm wondering if we could rationalize this a bit? Could put off to another PR, but wanted to bring it up.

Reply:
I agree that it's confusing. I'm not entirely sure if's necessary. I'll just note that this PR didn't actually add it, I just moved the struct up to the top to live with the other structs, so I think it's best to hold off on changing it to another PR.

Reply:
This was part of the original PR I created for bin packing, I am working on the fix for this as part of the refactoring for this file.